### ISSUE TITLE
When use multi-nodes to train the model, training loss  rises abnormally

### ISSUE THREAD

**Issue by XuecaiHu** [haotian-liu/LLaVA#1264]
### Describe the issue

Issue:
When use multi-nodes to train the model, training loss  rises abnormally
Command:
```
#!/bin/bash

deepspeed --hostfile=scripts/v1_5/hostfile.txt llava/train/train_mem.py \
    --lora_enable True --lora_r 128 --lora_alpha 256 --mm_projector_lr 2e-5 \
    --deepspeed ./scripts/zero2.json \
    --model_name_or_path /path_to_data/LLaVa/Models/vicuna-13b-v1.5 \
    --version v1 \
    --data_path /path_to_data/LLaVa/Data/LLaVA-Instruct-150K/llava_v1_5_mix665k.json \
    --image_folder /path_to_data/LLaVa/Data/ \
    --vision_tower /path_to_data/LLaVa/Models/clip-vit-large-patch14-336/ \
    --pretrain_mm_mlp_adapter /path_to_data/trained_model/checkpoints/llava-v1.5-13b-pretrain//mm_projector.bin \
    --mm_projector_type mlp2x_gelu \
    --mm_vision_select_layer -2 \
    --mm_use_im_start_end False \
    --mm_use_im_patch_token False \
    --image_aspect_ratio pad \
    --group_by_modality_length True \
    --bf16 True \
    --output_dir /path_to_data/trained_model/checkpoints/llava-v1.5-13b-lora \
    --num_train_epochs 1 \
    --per_device_train_batch_size 4 \
    --per_device_eval_batch_size 4 \
    --gradient_accumulation_steps 2 \
    --evaluation_strategy "no" \
    --save_strategy "steps" \
    --save_steps 50000 \
    --save_total_limit 1 \
    --learning_rate 2e-4 \
    --weight_decay 0. \
    --warmup_ratio 0.03 \
    --lr_scheduler_type "cosine" \
    --logging_steps 1 \
    --tf32 True \
    --model_max_length 2048 \
    --gradient_checkpointing True \
    --dataloader_num_workers 4 \
    --lazy_preprocess True \
    --report_to "none" 2>&1 | tee  llava_v1.5_0311.log

```

Log: 
```
192.168.50.171:   total_norm_cuda = get_accelerator().FloatTensor([float(total_norm)])
192.168.50.170: {'loss': 1.2992, 'learning_rate': 1.282051282051282e-06, 'epoch': 0.0}
192.168.50.170: {'loss': 1.2859, 'learning_rate': 2.564102564102564e-06, 'epoch': 0.0}
192.168.50.170: {'loss': 1.2685, 'learning_rate': 3.846153846153847e-06, 'epoch': 0.0}
192.168.50.170: {'loss': 1.3461, 'learning_rate': 5.128205128205128e-06, 'epoch': 0.0}
192.168.50.170: {'loss': 1.3198, 'learning_rate': 6.41025641025641e-06, 'epoch': 0.0}
192.168.50.170: {'loss': 1.2998, 'learning_rate': 7.692307692307694e-06, 'epoch': 0.0}
192.168.50.170: {'loss': 1.3204, 'learning_rate': 8.974358974358976e-06, 'epoch': 0.0}
192.168.50.170: {'loss': 1.2842, 'learning_rate': 1.0256410256410256e-05, 'epoch': 0.0}
192.168.50.170: {'loss': 1.349, 'learning_rate': 1.153846153846154e-05, 'epoch': 0.0}
192.168.50.170: {'loss': 1.2635, 'learning_rate': 1.282051282051282e-05, 'epoch': 0.0}
192.168.50.170: {'loss': 1.2919, 'learning_rate': 1.4102564102564104e-05, 'epoch': 0.0}
192.168.50.170: {'loss': 1.2695, 'learning_rate': 1.5384615384615387e-05, 'epoch': 0.0}
192.168.50.170: {'loss': 1.2735, 'learning_rate': 1.6666666666666667e-05, 'epoch': 0.0}
192.168.50.170: {'loss': 1.2399, 'learning_rate': 1.794871794871795e-05, 'epoch': 0.0}
192.168.50.170: {'loss': 1.2478, 'learning_rate': 1.923076923076923e-05, 'epoch': 0.0}
192.168.50.170: {'loss': 1.2887, 'learning_rate': 2.0512820512820512e-05, 'epoch': 0.0}
192.168.50.170: {'loss': 1.2083, 'learning_rate': 2.1794871794871795e-05, 'epoch': 0.0}
192.168.50.170: {'loss': 1.227, 'learning_rate': 2.307692307692308e-05, 'epoch': 0.0}
192.168.50.170: {'loss': 1.2554, 'learning_rate': 2.435897435897436e-05, 'epoch': 0.0}
192.168.50.170: {'loss': 1.1559, 'learning_rate': 2.564102564102564e-05, 'epoch': 0.0}
192.168.50.170: {'loss': 1.2511, 'learning_rate': 2.6923076923076923e-05, 'epoch': 0.0}
192.168.50.170: {'loss': 1.2031, 'learning_rate': 2.8205128205128207e-05, 'epoch': 0.0}
192.168.50.170: {'loss': 1.1509, 'learning_rate': 2.948717948717949e-05, 'epoch': 0.0}
192.168.50.170: {'loss': 1.2433, 'learning_rate': 3.0769230769230774e-05, 'epoch': 0.0}
192.168.50.170: {'loss': 1.202, 'learning_rate': 3.205128205128206e-05, 'epoch': 0.0}
192.168.50.170: {'loss': 1.1598, 'learning_rate': 3.3333333333333335e-05, 'epoch': 0.01}
192.168.50.170: {'loss': 1.2317, 'learning_rate': 3.461538461538462e-05, 'epoch': 0.01}
192.168.50.170: {'loss': 1.1928, 'learning_rate': 3.58974358974359e-05, 'epoch': 0.01}
192.168.50.170: {'loss': 1.1584, 'learning_rate': 3.717948717948718e-05, 'epoch': 0.01}
192.168.50.170: {'loss': 1.1689, 'learning_rate': 3.846153846153846e-05, 'epoch': 0.01}
192.168.50.170: {'loss': 1.2189, 'learning_rate': 3.974358974358974e-05, 'epoch': 0.01}
192.168.50.170: {'loss': 1.142, 'learning_rate': 4.1025641025641023e-05, 'epoch': 0.01}
192.168.50.170: {'loss': 1.1915, 'learning_rate': 4.230769230769231e-05, 'epoch': 0.01}
192.168.50.170: {'loss': 1.151, 'learning_rate': 4.358974358974359e-05, 'epoch': 0.01}
192.168.50.170: {'loss': 1.0986, 'learning_rate': 4.4871794871794874e-05, 'epoch': 0.01}
192.168.50.170: {'loss': 1.1711, 'learning_rate': 4.615384615384616e-05, 'epoch': 0.01}
192.168.50.170: {'loss': 0.2248, 'learning_rate': 4.7435897435897435e-05, 'epoch': 0.01}
192.168.50.170: {'loss': 1.1816, 'learning_rate': 4.871794871794872e-05, 'epoch': 0.01}
192.168.50.170: {'loss': 1.1495, 'learning_rate': 5e-05, 'epoch': 0.01}
......
192.168.50.170: {'loss': 6.3472, 'learning_rate': 0.0001999475152949459, 'epoch': 0.04}
192.168.50.170: {'loss': 8.034, 'learning_rate': 0.00019994547742763934, 'epoch': 0.04}
192.168.50.170: {'loss': 9.2512, 'learning_rate': 0.00019994340075805725, 'epoch': 0.04}
192.168.50.170: {'loss': 9.6909, 'learning_rate': 0.00019994128528700583, 'epoch': 0.04}
192.168.50.170: {'loss': 11.2645, 'learning_rate': 0.00019993913101530635, 'epoch': 0.04}
192.168.50.170: {'loss': 11.2146, 'learning_rate': 0.00019993693794379525, 'epoch': 0.04}
192.168.50.170: {'loss': 10.875, 'learning_rate': 0.0001999347060733239, 'epoch': 0.04}
192.168.50.170: {'loss': 10.65, 'learning_rate': 0.00019993243540475877, 'epoch': 0.04}
192.168.50.170: {'loss': 10.288, 'learning_rate': 0.00019993012593898146, 'epoch': 0.04}
192.168.50.170: {'loss': 11.3354, 'learning_rate': 0.00019992777767688854, 'epoch': 0.04}
192.168.50.170: {'loss': 11.0331, 'learning_rate': 0.00019992539061939175, 'epoch': 0.04}
192.168.50.170: {'loss': 9.385, 'learning_rate': 0.00019992296476741776, 'epoch': 0.04}
192.168.50.170: {'loss': 11.1739, 'learning_rate': 0.00019992050012190843, 'epoch': 0.04}
192.168.50.170: {'loss': 8.9434, 'learning_rate': 0.00019991799668382058, 'epoch': 0.04}
192.168.50.170: {'loss': 8.43, 'learning_rate': 0.00019991545445412613, 'epoch': 0.04}
192.168.50.170: {'loss': 8.0869, 'learning_rate': 0.0001999128734338121, 'epoch': 0.04}
192.168.50.170: {'loss': 8.9871, 'learning_rate': 0.00019991025362388044, 'epoch': 0.04}
192.168.50.170: {'loss': 10.097, 'learning_rate': 0.00019990759502534834, 'epoch': 0.04}
192.168.50.170: {'loss': 10.3712, 'learning_rate': 0.00019990489763924797, 'epoch': 0.04}
192.168.50.170: {'loss': 7.6627, 'learning_rate': 0.00019990216146662646, 'epoch': 0.04}
192.168.50.170: {'loss': 8.6533, 'learning_rate': 0.00019989938650854616, 'epoch': 0.04}
192.168.50.170: {'loss': 8.7901, 'learning_rate': 0.00019989657276608437, 'epoch': 0.04}
192.168.50.170: {'loss': 9.1619, 'learning_rate': 0.00019989372024033352, 'epoch': 0.04}
192.168.50.170: {'loss': 9.0868, 'learning_rate': 0.000199890828932401, 'epoch': 0.04}
192.168.50.170: {'loss': 9.3259, 'learning_rate': 0.00019988789884340936, 'epoch': 0.04}
192.168.50.170: {'loss': 9.7941, 'learning_rate': 0.00019988492997449615, 'epoch': 0.04}
192.168.50.170: {'loss': 10.0934, 'learning_rate': 0.00019988192232681397, 'epoch': 0.05}
192.168.50.170: {'loss': 9.4644, 'learning_rate': 0.00019987887590153055, 'epoch': 0.05}
192.168.50.170: {'loss': 8.8681, 'learning_rate': 0.00019987579069982855, 'epoch': 0.05}
192.168.50.170: {'loss': 8.274, 'learning_rate': 0.00019987266672290575, 'epoch': 0.05}
192.168.50.170: {'loss': 8.434, 'learning_rate': 0.00019986950397197503, 'epoch': 0.05}
192.168.50.170: {'loss': 8.905, 'learning_rate': 0.00019986630244826426, 'epoch': 0.05}
192.168.50.170: {'loss': 9.5568, 'learning_rate': 0.00019986306215301637, 'epoch': 0.05}
192.168.50.170: {'loss': 9.7974, 'learning_rate': 0.00019985978308748937, 'epoch': 0.05}
192.168.50.170: {'loss': 10.1468, 'learning_rate': 0.00019985646525295632, 'epoch': 0.05}
192.168.50.170: {'loss': 9.6746, 'learning_rate': 0.0001998531086507053, 'epoch': 0.05}
192.168.50.170: {'loss': 9.2765, 'learning_rate': 0.00019984971328203946, 'epoch': 0.05}

```

Screenshots:
You may attach screenshots if it better explains the issue.

**Comment by manushree635** [comment]
were you able to resolve this

**Comment by fisher75** [comment]
Hi, how do you know the training was effecitve? Did you use the default training setting? I LoRA with default parameters and basically no improvement.
